From 3e08af573f8b50da5202f55df62c925f88a47f3d Mon Sep 17 00:00:00 2001
From: namannigam <namannigam12@gmail.com>
Date: Thu, 10 Aug 2017 08:45:27 +0530
Subject: [PATCH] LANG-1349: EqualsBuilder#isRegistered: swappedPair
 construction bug (closes #282)

SwappedPair constructed as Pair.of(rhs,lhs)
---
 .../java/org/apache/commons/lang3/builder/EqualsBuilder.java    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/java/org/apache/commons/lang3/builder/EqualsBuilder.java b/src/main/java/org/apache/commons/lang3/builder/EqualsBuilder.java
index 4e24bc5248c..d2cf7c7fb5e 100644
--- a/src/main/java/org/apache/commons/lang3/builder/EqualsBuilder.java
+++ b/src/main/java/org/apache/commons/lang3/builder/EqualsBuilder.java
@@ -157,7 +157,7 @@ static Pair<IDKey, IDKey> getRegisterPair(final Object lhs, final Object rhs) {
     static boolean isRegistered(final Object lhs, final Object rhs) {
         final Set<Pair<IDKey, IDKey>> registry = getRegistry();
         final Pair<IDKey, IDKey> pair = getRegisterPair(lhs, rhs);
-        final Pair<IDKey, IDKey> swappedPair = Pair.of(pair.getLeft(), pair.getRight());
+        final Pair<IDKey, IDKey> swappedPair = Pair.of(pair.getRight(), pair.getLeft());
 
         return registry != null
                 && (registry.contains(pair) || registry.contains(swappedPair));
